numpy/core/src/multiarray/arrayfunction_override.c

```diff
@@ -348,30 +342,35 @@ array_implement_array_function(
      */
     if (kwargs != NULL && PyDict_Contains(kwargs, npy_ma_str_like)) {
         PyObject *like_arg = PyDict_GetItem(kwargs, npy_ma_str_like);
-        if (like_arg != NULL) {
+        if (like_arg != NULL && like_arg != Py_None) {
```

Comment:
Thanks for the updates, this exactly what I was hoping for!  Some of the reformatting looks a bit unnecessary, but maybe automated (I don't care much if it stays).

Here, I think `like=None` should never happen and that is fairly clear now.  Maybe just add an `assert(like != Py_None)` somewhere and undo these changes?

Reply:
@seberg addressed in 4bfce86. Hopefully all good now?

Reply:
Well, half the reason for it is that I preferred fixing it one level higher in the interface code, rather than the dispatching helpers.

Do you mind if I just reset the dispatching changes back to the original state (and back in the small comment change, since IIRC that is correct)?